### PR TITLE
MAINT: Remove Duplicated Code (function extract rmap)

### PR DESCRIPTION
**Pull request by archsyscall**
Duplicate was removed by extracting a function from `linspace` that return `rmap`.

Thank you!

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
I'd prefer a list-comprehension to a lambda function, e.g.

```python
>>> left_ramp, right_ramp = [
...    np.linspace(
...        start=end_value_pair[i]
...        stop=edge_pair[i].squeeze(axis)
...        num=width_pair[i],
...        endpoint=False
...        dtype=padded.dtype,
...        axis=axis
...    ) for i in range(2)
... ]
```

**Comment by marload** [comment]
The method is good, but I personally think it's better to use lambda to make it a function. I want to hear other opinions!

**Comment by eric-wieser** [comment]
I was thinking the same as @rossbar, although I'd write it:
```python
left_ramp, right_ramp = (
    np.linspace(
        start=end_value
        stop=edge.squeeze(axis)
        num=width,
        endpoint=False
        dtype=padded.dtype,
        axis=axis
    ) for end_value, edge, width in zip(end_value_pair, edge_pair, width_pair)
)
```

File: numpy/lib/arraypad.py [hunk 1]
```diff
@@ -207,23 +207,17 @@ def _get_linear_ramps(padded, axis, width_pair, end_value_pair):
     """
     edge_pair = _get_edges(padded, axis, width_pair)
 
-    left_ramp = np.linspace(
-        start=end_value_pair[0],
-        stop=edge_pair[0].squeeze(axis),  # Dimensions is replaced by linspace
-        num=width_pair[0],
+    rmap = lambda x: np.linspace(
```
Comment (eric-wieser) [inline numpy/lib/arraypad.py:210]: typo, `ramp`

**Comment by marload** [comment]
I reflected on you review! Thank you for your comments!

File: numpy/lib/arraypad.py [hunk 2]
```diff
@@ -207,23 +207,18 @@ def _get_linear_ramps(padded, axis, width_pair, end_value_pair):
...
-    )
-
-    right_ramp = np.linspace(
-        start=end_value_pair[1],
-        stop=edge_pair[1].squeeze(axis),  # Dimension is replaced by linspace
-        num=width_pair[1],
-        endpoint=False,
-        dtype=padded.dtype,
-        axis=axis,
+        axis=axis
```
Comment (eric-wieser) [inline numpy/lib/arraypad.py:217]: Can you indent all the lines between the opening `(` and this one, as I do in my comment?

**Review by rossbar — APPROVED** [review]
LGTM - there's potential for some additional style touch-ups but it's fine as-is. Thanks @marload !

**Comment by mattip** [comment]
Thanks @marload

**Pull request merged**